test/boost/schema_loader_test.cc

```diff
@@ -277,3 +283,54 @@ SEASTAR_THREAD_TEST_CASE(test_mv_index) {
                     "CREATE INDEX ON ks.cf (v1);").get(),
             {view_type::view, view_type::index, view_type::view, view_type::index});
 }
+
+void check_sstable_schema(sstables::test_env& env, std::filesystem::path sst_path, const std::vector<mutation>& mutations) {
+    db::config dbcfg;
+
+    auto schema = tools::load_schema_from_sstable(dbcfg, sst_path).get();
+
+    const auto ed = sstables::parse_path(sst_path, "ks", "tbl");
+    const auto dir_path = sst_path.parent_path();
+    data_dictionary::storage_options local;
+    auto sst = env.manager().make_sstable(schema, dir_path.c_str(), local, ed.generation, sstables::sstable_state::normal, ed.version, ed.format);
+
+    sst->load(schema->get_sharder()).get();
+
+    auto rd = assert_that(sst->make_reader(schema, env.make_reader_permit(), query::full_partition_range, schema->full_slice()));
+    for (const auto& m : mutations) {
+        rd.produces(m);
+    }
+    rd.produces_end_of_stream();
+}
+
+SEASTAR_TEST_CASE(test_load_schema_from_sstable) {
+    return sstables::test_env::do_with_async([&] (sstables::test_env& env) {
+        auto random_spec = tests::make_random_schema_specification(
+                get_name(),
+                std::uniform_int_distribution<size_t>(1, 4), // partition-key
+                std::uniform_int_distribution<size_t>(0, 4), // clustering-key
+                std::uniform_int_distribution<size_t>(0, 8), // static
+                std::uniform_int_distribution<size_t>(0, 8)); // regular
+        auto random_schema = tests::random_schema{tests::random::get_int<uint32_t>(), *random_spec};
+        auto schema = random_schema.schema();
+
```

Comment:
Should we test collections here? Are their encodings different?

Reply:
Or maybe random_schema already does?

Also, what about UDTs?

Reply:
Yes, random schema uses both UDTs and collections as key types. I am 100% certain because the test failed as soon as it hit a UDT and I had to fix my parsing code to work with it.